scripts/testing/database-emulator-test.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const request = require('request');
+const tmp = require('tmp');
+
+const { spawn } = require('child-process-promise');
+
+const EMULATOR_LINK =
+  'https://storage.googleapis.com/firebase-preview-drop/emulator/firebase-database-emulator-v3.5.0.jar';
+const EMULATOR_FILENAME = 'database-emulator.jar';
+
+const EMULATOR_PORT = 8088;
+const EMULATOR_NAMESPACE = 'test-emulator';
+
+async function delay(seconds) {
+  return new Promise(resolve => setTimeout(resolve, 1000 * seconds));
+}
+
+async function downloadEmulator() {
+  return new Promise((resolve, reject) => {
+    tmp.dir((err, dir) => {
+      if (err) reject(err);
+      console.log(`Created temporary directory at [${dir}].`);
+      const filepath = path.resolve(dir, EMULATOR_FILENAME);
+      const writeStream = fs.createWriteStream(filepath);
+      console.log(`Downloading emulator from [${EMULATOR_LINK}] ...`);
+      request(EMULATOR_LINK)
+        .pipe(writeStream)
+        .on('finish', () => {
+          resolve(filepath);
+          console.log(`Saved emulator binary file to [${filepath}].`);
```

Comment:
nit: I'd do the console.log() before resolving the promise.

Reply:
Done.